source/uwp/Renderer/lib/XamlBuilder.cpp

```diff
@@ -1123,6 +1123,243 @@ namespace AdaptiveCards { namespace Rendering { namespace Uwp
         THROW_IF_FAILED(textConfig->get_MaxWidth(&maxWidth));
         StyleXamlTextBlock(textSize, textColor, containerStyle, Boolify(isSubtle), wrap, maxWidth, textWeight, xamlTextBlock, hostConfig);
     }
+    
+    HRESULT GetTextFromXmlNode(
+        ABI::Windows::Data::Xml::Dom::IXmlNode* node,
+        HSTRING* text)
+    {
+        ComPtr<ABI::Windows::Data::Xml::Dom::IXmlNode> localNode = node;
+        ComPtr<ABI::Windows::Data::Xml::Dom::IXmlNodeSerializer> textNodeSerializer;
+        RETURN_IF_FAILED(localNode.As(&textNodeSerializer));
+
+        RETURN_IF_FAILED(textNodeSerializer->get_InnerText(text));
+        return S_OK;
+    }
+
+    HRESULT XamlRunFromXmlNode(
+        ABI::Windows::Data::Xml::Dom::IXmlNode* node,
+        ABI::Windows::UI::Xaml::Documents::IRun** run)
+    {
+        HString text;
+        RETURN_IF_FAILED(GetTextFromXmlNode(node, text.GetAddressOf()));
+
+        ComPtr<ABI::Windows::UI::Xaml::Documents::IRun> localRun = XamlHelpers::CreateXamlClass<ABI::Windows::UI::Xaml::Documents::IRun>(HStringReference(RuntimeClass_Windows_UI_Xaml_Documents_Run));
+        RETURN_IF_FAILED(localRun->put_Text(text.Get()));
+
+        *run = localRun.Detach();
+        return S_OK;
+    }
+
+    HRESULT AddListInlines(
+        ABI::Windows::Data::Xml::Dom::IXmlNode* node,
+        bool isListOrdered,
+        IVector<ABI::Windows::UI::Xaml::Documents::Inline*>* inlines)
+    {
+        ComPtr<ABI::Windows::Data::Xml::Dom::IXmlNode> listChild;
+        RETURN_IF_FAILED(node->get_FirstChild(&listChild));
+
+        UINT iteration = 1;
+        while (listChild != nullptr)
+        {
+            HString text;
+            RETURN_IF_FAILED(GetTextFromXmlNode(listChild.Get(), text.GetAddressOf()));
+
+            std::wstring listElementString = L"\n";
+            if (!isListOrdered)
+            {
+                listElementString += L"● ";
+            }
+            else
+            {
+                wchar_t buffer[10];
+                swprintf_s(buffer, ARRAYSIZE(buffer), L"%d. ", iteration);
+
+                std::wstring numberElementString(buffer);
+                listElementString += numberElementString;
+            }
+            listElementString += text.GetRawBuffer(nullptr);
+
+            HString listElementHString;
+            RETURN_IF_FAILED(WindowsCreateString(listElementString.c_str(), listElementString.length(), listElementHString.GetAddressOf()));
```

Comment:
>WindowsCreateString [](start = 29, length = 19)

Would listElementHString.Set(listElementString.c_str()) work instead of calling WIndowsCreateString?